pages/_app.js

```diff
@@ -2,17 +2,142 @@ import GlobalStyle from "../styles";
 import { useImmerLocalStorageState } from "@/hooks/useImmerLocalStorageState";
 import { contactsSampleData } from "@/data/SampleData";
 import Layout from "@/components/Layout/Layout";
+import { uid } from "uid";
+import { useRouter } from "next/router";
 
 export default function App({ Component, pageProps }) {
+  const router = useRouter();
   const [contacts, updateContacts] = useImmerLocalStorageState("contacts", {
```

Comment:
I am not sure why and how do you need UseImmer in your project, and if its the best option for the project in this moment. 
If you need to use Localstorage, we suggest the `use-local-storage-state` hook, if you need a state management, we suggest `zustand` (but I dont see that necessary at this point) 

please rethink if `Immer` is the best choice .

Reply:
I removed the hook useImmerLocalStorageState and used useLocalStorageState instead